core/matrix/csr.cpp

```diff
@@ -402,6 +403,48 @@ std::unique_ptr<LinOp> Csr<ValueType, IndexType>::conj_transpose() const
 }
 
 
+template <typename ValueType, typename IndexType>
+std::unique_ptr<LinOp> Csr<ValueType, IndexType>::permute(
+    const Array<IndexType> *permutation_indices) const
+{
+    GKO_ASSERT_IS_SQUARE_MATRIX(this);
+    GKO_ASSERT_EQ(permutation_indices->get_num_elems(), this->get_size()[0]);
+    auto exec = this->get_executor();
+    auto permute_cpy =
+        Csr::create(exec, this->get_size(), this->get_num_stored_elements(),
+                    this->get_strategy());
+    Array<IndexType> inv_permutation(exec, this->get_size()[1]);
+
+    exec->run(csr::make_invert_permutation(
+        this->get_size()[1],
+        make_temporary_clone(exec, permutation_indices)->get_const_data(),
+        inv_permutation.get_data()));
+    exec->run(csr::make_inv_symm_permute(inv_permutation.get_const_data(), this,
+                                         permute_cpy.get()));
```

Comment:
In practice, do we use inv_sym_permute more than sym_permute?

Reply:
inv_symm_permute or inv_sym_permute?
I think "symm" is good as it avoids confusion with "symbolic"

Reply:
The reason we have a inv_symm_permute kernel, but not a symm_permute_kernel is that for the column permutation part, we need the inverse permutation to to
```
new_column[i] = inv_perm[old_column[i]]
```
due to the sparse nature of the operation, it doesn't work the other way round.